src/main/scala/mesosphere/marathon/io/IO.scala

```diff
@@ -47,21 +45,19 @@ object IO extends StrictLogging {
     * @param mabyeTo Outputstream to copy to.
     * @return
     */
-  def transfer(maybeFrom: Option[InputStream], mabyeTo: Option[OutputStream]): Long = {
-    (maybeFrom, mabyeTo) match {
+  def transfer(maybeFrom: Option[InputStream], maybeTo: Option[OutputStream]): Long = {
+    (maybeFrom, maybeTo) match {
       case (Some(from), Some(to)) =>
-        val buf = createBuffer();
-        var total = 0
-        while (true) {
-          val r = from.read(buf)
-          if (r == -1) {
-            break
+        @tailrec def iter(buf: Array[Byte], total: Long): Long =
+          from.read(buf) match {
+            case -1 => total
+            case r =>
+              to.write(buf, 0, r)
+              to.flush()
+              iter(buf, total + r)
           }
-          to.write(buf, 0, r)
-          to.flush()
-          total += r
-        }
-        total
+
+        iter(new Array[Byte](8192), 0L)
```

Comment:
Would it make sense to actually make the buffer size larger?

Reply:
This is the default from Guava and Apache Commons IO. So I'm not sure if it makes sense to differ from them.

Reply:
Thank you!